src/rabbitmq/message_consumer.rs

```diff
@@ -212,3 +246,55 @@ impl ManualAck for Option<Ackable> {
         }
     }
 }
+
+#[cfg(feature = "telemetry")]
```

Comment:
Instead of putting a lot of `#[cfg(feature)]` guards, maybe it's clearer to do this instead?

```rust
#[cfg(feature = "telemetry")]
mod telemetry {
...
}

#[cfg(feature = "telemetry")]
pub use telemetry::*;
```